konnectivity-client/pkg/client/client_test.go

```diff
@@ -151,6 +152,15 @@ func TestClose(t *testing.T) {
 	}
 }
 
+func TestCreateSingleUseGrpcTunnel_NoLeakOnFailure(t *testing.T) {
+	defer goleak.VerifyNone(t, goleak.IgnoreCurrent())
+
+	_, err := CreateSingleUseGrpcTunnel(context.Background(), "127.0.0.1:12345", grpc.WithInsecure())
+	if err == nil {
```

Comment:
```suggestion
        // Test a connection that fails to establish
	t, err := CreateSingleUseGrpcTunnel(context.Background(), "127.0.0.1:12345", grpc.WithInsecure())
	if err == nil || t != nil {
```

Reply:
Untested, but just looking at the code seems we could test if `t` is nil or not.

My main suggestion here, though, is not that but to have a simple comment saying we expect this connection to fail (another option might be to have a var for the IP/port and use a name that makes it clear we want this to fail)